Remove ExtLlib dependency

shonfeder opened the pull request:
Closes #32

Tests are passing and everything builds locally for me, but I haven't yet done any integration testing. I'll pin this branch for a project I'm working on tomorrow and see if it works as expected.

I trust you'll let me know what additional checks I should perform or tests I should add, and if I can improve my approach on any details. :)

astrada committed:
Update version to 0.3.20

shonfeder committed:
Gitignore _opam dir

shonfeder committed:
s/Std.identity/Fun.id

shonfeder committed:
Replace Option.map_default

shonfeder committed:
Replace Option.default with Option.value

shonfeder committed:
Replace Option.may with Option.iter

shonfeder committed:
Remove calls to ExtString

shonfeder committed:
Remove use of ExtList

shonfeder committed:
Remove extlib dependency

shonfeder committed:
Correct location of doc comment

astrada commented:
Thanks! I'm going to test it against [google-drive-ocamlfuse](https://github.com/astrada/google-drive-ocamlfuse).

astrada committed:
Fix strip_string

Add '\r' to stripped characters.
https://github.com/ygrek/ocaml-extlib/blob/master/src/extString.mli#L139

astrada commented on src/gapi/gapiUtils.ml:
```diff
@@ -77,6 +77,37 @@ let string_before_char c s =
+
+let strip_string s =
+  let leading_trailing_whitespace =
+    Str.regexp "\\(^[ \t\n]+\\|[ \t\n]+$\\)"
```
I fixed a little bug here, adding `\r` to the regex.

shonfeder replied:
Thanks for catching that :)

astrada commented:
So far, it looks good. I'm going to test it a little bit more.

shonfeder commented:
Just reporting that I tested in a little project exercising a (tiny) part of the gmail API, and it's still working as I expected. :)

shonfeder commented:
Nice! Now a bunch of useful libs can be use along side.

 Thanks for your help here :)

Let me know if you could use any help in cutting a new release to the opam repo.

astrada commented:
I've already created the PR (ocaml/opam-repository#16632), unfortunately the CI is not working properly: Travis says the generated logs are too long, and the OCaml CI in the failed builds says that the docker image is unavailable.

astrada commented:
Actually I've closed that and reopened a new one (ocaml/opam-repository#16635) because I fixed another bug.
